pkg/doc/jose/encrypter.go

```diff
@@ -0,0 +1,188 @@
+	}
+
+	kt, err := ecdhes.ECDHES256KWAES256GCMKeyTemplateWithRecipients(recipientsPubKeys)
+	if err != nil {
+		return nil, err
+	}
+
+	senderKH, err := keyset.NewHandle(kt)
+	if err != nil {
+		return nil, err
+	}
+
+	return &JWEEncrypt{
+		recipients: recipientsPubKeys,
+		senderKH:   senderKH,
+	}, nil
+}
+
+func (je *JWEEncrypt) getEncryptionPrimitive() (api.CompositeEncrypt, error) {
+	senderPubKH, err := je.senderKH.Public()
+	if err != nil {
+		return nil, err
+	}
+
+	return ecdhes.NewECDHESEncrypt(senderPubKH)
+}
+
+// Encrypt plaintext with aad and return a JSONWebEncryption instance to serialize JWE instance
+func (je *JWEEncrypt) Encrypt(plaintext, aad []byte) (*JSONWebEncryption, error) {
+	encPrimitive, err := je.getEncryptionPrimitive()
+	if err != nil {
+		return nil, err
+	}
+
+	protectedHeaders := map[string]interface{}{
+		HeaderEncryption: subtle.A256GCM, // TODO add support for Chacha content encryption
+	}
+
+	// TODO - Go jose adds CEK as part of protectedHeaders, see if this is valid. Also, for a single recipient,
+	//  Go jose merges recipient header into JWE protect protectedHeaders. See if this is needed too.
+
+	authData, err := computeAuthData(protectedHeaders, aad)
+	if err != nil {
+		return nil, err
+	}
+
+	serializedEncData, err := encPrimitive.Encrypt(plaintext, authData)
+	if err != nil {
+		return nil, err
+	}
+
+	encData := new(subtle.EncryptedData)
+
+	err = json.Unmarshal(serializedEncData, encData)
+	if err != nil {
+		return nil, err
+	}
+
+	var recipients []Recipient
+
+	for _, rec := range encData.Recipients {
+		var mRecJWK []byte
+
+		mRecJWK, err = convertRecKeyToMarshalledJWK(rec)
+		if err != nil {
+			return nil, err
+		}
+
+		recipients = append(recipients, Recipient{
+			EncryptedKey: string(rec.EncryptedCEK),
+			Header: RecipientHeaders{
+				ALG: rec.Alg,
+				EPK: string(mRecJWK),
+			},
+		})
+	}
+
+	jsonEncryption := &JSONWebEncryption{
+		IV:               string(encData.IV),
+		Tag:              string(encData.Tag),
+		Ciphertext:       string(encData.Ciphertext),
+		Recipients:       recipients,
+		ProtectedHeaders: protectedHeaders,
+	}
+
+	return jsonEncryption, nil
+}
+
+func convertRecKeyToMarshalledJWK(rec *subtle.RecipientWrappedKey) ([]byte, error) {
+	var c elliptic.Curve
+
+	c, err := hybrid.GetCurve(rec.EPK.Curve)
+	if err != nil {
+		return nil, err
+	}
+
+	recJWK := JWK{
+		JSONWebKey: jose.JSONWebKey{
+			Use: HeaderEncryption,
+			Key: &ecdsa.PublicKey{
+				Curve: c,
+				X:     new(big.Int).SetBytes(rec.EPK.X),
+				Y:     new(big.Int).SetBytes(rec.EPK.Y),
+			},
+		},
+		Kty: "EC", // TODO add support for Chacha content encryption
+		Crv: rec.EPK.Curve,
+	}
+
+	return recJWK.MarshalJSON()
+}
+
+// Decrypt serializedJWE by first deserializing it, then decrypting the underlying JWE instance and return plaintext
+func (je *JWEEncrypt) Decrypt(serializedJWE []byte) ([]byte, error) {
+	return nil, fmt.Errorf("TODO - implement me")
```

Comment:
You can link my story https://github.com/hyperledger/aries-framework-go/issues/1507 here

Reply:
current issue is not closed yet... will implement as a followup to this PR with the same issue

Reply:
added link in comment